sklearn/preprocessing/tests/test_data.py

```diff
@@ -189,19 +209,338 @@ def test_scaler_2d_arrays():
+    # Test if partial_fit run over many batches of size 1 and 50
+    # gives the same results as fit
+
+    for chunk_size in [1, 50]:
+
+        rng = np.random.RandomState(0)
+        n_features = 50
+        n_samples = 20
+        offsets = rng.uniform(-1, 1, size=n_features)
+        scales = rng.uniform(1, 10, size=n_features)
+
+        data_chunks = [rng.randn(chunk_size, n_features) * scales + offsets
+                       for i in range(n_samples)]
+
+        # Test mean at the end of the process
+        X = np.vstack(data_chunks)
+        scaler_batch = MinMaxScaler().fit(X)
+
+        scaler_incr = MinMaxScaler()
+        for data_chunk in data_chunks:
+            scaler_incr = scaler_incr.partial_fit(data_chunk)
+
+        assert_array_almost_equal(scaler_batch.data_min_,
+                                  scaler_incr.data_min_)
+        assert_array_almost_equal(scaler_batch.data_max_,
+                                  scaler_incr.data_max_)
+        assert_equal(scaler_batch.n_samples_seen_, scaler_incr.n_samples_seen_)
+        assert_array_almost_equal(scaler_batch.data_range_,
+                                  scaler_incr.data_range_)
+        assert_array_almost_equal(scaler_batch.scale_, scaler_incr.scale_)
+        assert_array_almost_equal(scaler_batch.min_, scaler_incr.min_)
+
+        # Test std after 1 step
+        scaler_batch = MinMaxScaler().fit(data_chunks[0])
+        scaler_incr = MinMaxScaler().partial_fit(data_chunks[0])
+
+        assert_array_almost_equal(scaler_batch.data_min_,
+                                  scaler_incr.data_min_)
+        assert_array_almost_equal(scaler_batch.data_max_,
+                                  scaler_incr.data_max_)
+        assert_equal(scaler_batch.n_samples_seen_, scaler_incr.n_samples_seen_)
+        assert_array_almost_equal(scaler_batch.data_range_,
+                                  scaler_incr.data_range_)
+        assert_array_almost_equal(scaler_batch.scale_, scaler_incr.scale_)
+        assert_array_almost_equal(scaler_batch.min_, scaler_incr.min_)
+
+        # Test std until the end of partial fits, and
+        X = np.vstack(data_chunks)
+        scaler_batch = MinMaxScaler().fit(X)
+        scaler_incr = MinMaxScaler()  # Clean estimator
+        for i, data_chunk in enumerate(data_chunks):
+            scaler_incr = scaler_incr.partial_fit(data_chunk)
+            assert_equal((i+1) * chunk_size, scaler_incr.n_samples_seen_)
+
+
+def test_standard_scaler_partial_fit():
+    # Test if partial_fit run over many batches of size 1 and 50
+    # gives the same results as fit
+
+    for chunk_size in [1, 50]:
+        rng = np.random.RandomState(0)
+        n_features = 50
+        n_samples = 20
+        offsets = rng.uniform(-1, 1, size=n_features)
+        scales = rng.uniform(1, 10, size=n_features)
+
+        data_chunks = [rng.randn(chunk_size, n_features) * scales + offsets
+                       for i in range(n_samples)]
```

Comment:
I think this test would be easier to follow if the data matrix `X` was randomly generated once and for all at the top of the test function and then you would use

```python
for batch in gen_batches(n_samples, chunk_size):
    scaler_incr.partial_fit(X[batch])
```

whenever you need to iterate over batches. You can import `gen_batches` from `sklearn.utils`.